.github/workflows/update-db-csv-files.yml

```diff
@@ -0,0 +1,70 @@
+name: Update DB files
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Install system dependencies
+        run: |
+          sudo add-apt-repository ppa:deadsnakes/ppa
+          sudo apt-get update
+          sudo apt-get install -y pip python3.10-venv libcurl4-openssl-dev
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Create Google Service credentials file
+        uses: jsdaniell/create-json@v1.2.3
+        with:
+          name: "service_account.json"
+          json: ${{ secrets.GC_JSON }}
+
+      - name: Update dump files
+        run: |
+          python3 -m pip install --upgrade pip
+          pip install gspread pandas numpy
+          python3 apps/openchallenges/db-update/update_db_csv.py
+
+      # - name: Install schematic and validate files
+      #   shell: bash
+      #   run: |
+      #     python3.10 -m venv .venv
+      #     chmod 755 .venv/bin/activate
+      #     source .venv/bin/activate
+      #     pip3.10 install schematicpy
+
+      - name: Commit files
+        run: |
+          git config user.name github-actions[bot]
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+          git config advice.addIgnoredFile false
+          git add .
+          git commit -m "add latest CSV dump files for DB update"
```

Comment:
Based on the image shared in the preview, it looks like the commit in made on your behalf. We should ultimately use a service account. This can be done in another PR.

Reply:
The commit is done by the github-bot account, but you are correct in that the push and PR are opened by me.  FWIW though, the screenshots are taken from my sandbox repo, where I used my account's token for `GITHUB_TOKEN`. I believe for the monorepo, you have configured GITHUB_TOKEN to use a bot, right?

If not, then happy to create a service account for the OC!